internal/peering-request-operator/peering-request-controller.go

```diff
@@ -75,19 +77,27 @@ func (r *PeeringRequestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 		return ctrl.Result{RequeueAfter: r.retryTimeout}, err
 	}
 
-	exists, err := r.BroadcasterExists(pr)
-	if err != nil {
-		klog.Error(err, err.Error())
-		return ctrl.Result{RequeueAfter: r.retryTimeout}, err
+	exists := pr.Status.BroadcasterRef != nil
+	if exists {
+		// check if it really exists
+		exists, err = r.BroadcasterExists(pr)
```

Comment:
If the deployment does not exist, do we erase the reference somewhere?

Reply:
No, the only two ways that the deployment does not exist are:
* the PeeringRequest is just been created, so the reference is not set
* someone has deleted the deployment manually, so we recreate it immediately and we overwrite the reference